### PR TITLE
Implementation of the WASM wrapper for the XChaCha20Poly1305 algorithm. 

### PR DESCRIPTION
**Pull request by dumitory-dev**
I decided not to split the buffer into chunks, since the algorithm can handle quite large files.
"The maximum message size that can be encrypted using XChaCha20Poly1305 is 2^64 - 1 bytes, or approximately 18.4 exabytes "(EB)"

### CODE REVIEW

**Comment by OlegMell** [comment]
Great! magnificent code

**Comment by dumitory-dev** [comment]
> Great! magnificent code

Please approve

**Review by OlegMell — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+/// The ciphertext.
+///
+/// # Errors
+///
+/// Returns an error if the plaintext is empty or the key is not 32 bytes long.
+#[wasm_bindgen]
+pub fn encrypt_async(plaintext: Vec<u8>, key: Vec<u8>) -> js_sys::Promise {
+    // Wrap Rust future in a JavaScript Promise
+    future_to_promise(async move {
+        validate_key_size(&key)?;
```
Comment (slonikappa) [inline src/lib.rs:97]: will the method execution be aborted if Err result is returned in validation methods ?

**Review by slonikappa — APPROVED** [review]
Single comment so the review is approved anyway even if a fix will be necessary

**Pull request merged**